DANS-KNAW/dd-workflow-step-add-nbn-link
# DD-777 Template for description text

Fixes DD-777

# Description of changes
Implements the basic behavior:

* The service checks if the link is already present, usign the marker.
* If it is already present, it resumes with Success, with no changes.
* If it is not found, it adds the description element, concatenating the marker at the end. The marker should make it easier to bulk delete the description elements from the database after the mirroring is no longer necessary.
* If the service fails it will resume with Failure status, which will cause the dataset version to stay at `:draft`.
* No support added (yet) for fixing datasets that fail, such as retry mechanisms.

# How to test
* Deploy the service and configure it:
```
cd dd-tap
deploy.py -m dd-workflow-step-add-nbn-link -t configure-prepublish-workflow
```
Create some dataset versions and check that the link is added, and added only once per dataset.


# Notify
@DANS-KNAW/dataversedans
